packages/runtime/src/useCases/consumption/attributes/ChangeDefaultRepositoryAttribute.ts

```diff
@@ -26,6 +26,10 @@ export class ChangeDefaultRepositoryAttributeUseCase extends UseCase<ChangeDefau
     }
 
     protected async executeInternal(request: ChangeDefaultRepositoryAttributeRequest): Promise<Result<LocalAttributeDTO>> {
+        if (!this.attributesController.parent.consumptionConfig.setDefaultRepositoryAttributes) {
+            return Result.fail(RuntimeErrors.attributes.setDefaultRepositoryAttributesIsDisabled());
+        }
+
```

Comment:
do we really need this?

1. I am not sure if I like the access to the ConsumptionController to the config here
2. this will fail in line 40 with a good enough error
3. Point 2 should be enough as it's very unlikely that someone who has configured `setDefaultRepositoryAttributes` to false accesses that usecase.

Reply:
if you agree, don't forget to also remove the error from `packages/runtime/src/useCases/common/RuntimeErrors.ts`

Reply:
Not sure if I would agree with your 3. point, also given that false is the default setting. ;) In this case I think it is more appropriate to return this error before one of the other errors can be returned. Otherwise it might be frustrating figuring out what you did wrong, fixing it, and then getting the error that you cannot use this use case in the first place. And I don't see how it hurts here.

Reply:
wtb not setting a default?

Reply:
What do you mean by that?

Reply:
let's have a ☎ about this